v2/src/log_sink_etw.c

```diff
@@ -0,0 +1,497 @@
+                    /* Codes_SRS_LOG_SINK_ETW_01_064: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_int64_t, a byte with the value TlgInINT64 shall be added in the metadata. ]*/
+                    *pos = TlgInINT64;
+                    break;
+                case LOG_CONTEXT_PROPERTY_TYPE_uint64_t:
+                    /* Codes_SRS_LOG_SINK_ETW_01_065: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_uint64_t, a byte with the value TlgInUINT64 shall be added in the metadata. ]*/
+                    *pos = TlgInUINT64;
+                    break;
+                case LOG_CONTEXT_PROPERTY_TYPE_int32_t:
+                    /* Codes_SRS_LOG_SINK_ETW_01_069: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_int32_t, a byte with the value TlgInINT32 shall be added in the metadata. ]*/
+                    *pos = TlgInINT32;
+                    break;
+                case LOG_CONTEXT_PROPERTY_TYPE_uint32_t:
+                    /* Codes_SRS_LOG_SINK_ETW_01_071: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_uint32_t, a byte with the value TlgInUINT32 shall be added in the metadata. ]*/
+                    *pos = TlgInUINT32;
+                    break;
+                case LOG_CONTEXT_PROPERTY_TYPE_int16_t:
+                    /* Codes_SRS_LOG_SINK_ETW_01_073: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_int16_t, a byte with the value TlgInINT16 shall be added in the metadata. ]*/
+                    *pos = TlgInINT16;
+                    break;
+                case LOG_CONTEXT_PROPERTY_TYPE_uint16_t:
+                    /* Codes_SRS_LOG_SINK_ETW_01_075: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_uint16_t, a byte with the value TlgInUINT16 shall be added in the metadata. ]*/
+                    *pos = TlgInUINT16;
+                    break;
+                case LOG_CONTEXT_PROPERTY_TYPE_int8_t:
+                    /* Codes_SRS_LOG_SINK_ETW_01_077: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_int8_t, a byte with the value TlgInINT8 shall be added in the metadata. ]*/
+                    *pos = TlgInINT8;
+                    break;
+                case LOG_CONTEXT_PROPERTY_TYPE_uint8_t:
+                    /* Codes_SRS_LOG_SINK_ETW_01_079: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_uint8_t, a byte with the value TlgInUINT8 shall be added in the metadata. ]*/
+                    *pos = TlgInUINT8;
+                    break;
+
+                case LOG_CONTEXT_PROPERTY_TYPE_struct:
+                    /* Codes_SRS_LOG_SINK_ETW_01_052: [ For struct properties one extra byte shall be added for the field count. ]*/
+                    /* Codes_SRS_LOG_SINK_ETW_01_056: [ If the property type is LOG_CONTEXT_PROPERTY_TYPE_struct, a byte with the value _TlgInSTRUCT | _TlgInChain shall be added in the metadata. ]*/
+                    *pos = _TlgInSTRUCT | _TlgInChain;
+                    pos++;
+
+                    if (pos == &self_described_event->metadata[MAX_METADATA_SIZE])
```

Comment:
>self_described_event [](http://example.com/codeflow?start=32&length=20)

If line 172 executes and line 229 executes, doesn't that leave the serialized structure in an inconsistent state?

Reply:
We'll reset to not using any properties if we hit this error